File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let answer = 0;
    let isFinish = false;
    const vowels = ['A', 'E', 'I', 'O', 'U'];
    
    function dfs (curStr) {
        if (curStr.length > 5 || isFinish) {
            return;
        }
        
        if (curStr === word) {
            isFinish = true;
            return;
        }
        answer += 1;
        for (let i = 0; i < vowels.length; i++) {
            dfs(curStr + vowels[i]);
        }
    }
    dfs('');
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from collections import deque
def solution(word):
    dictionary = []
    vowels = ['A', 'E', 'I', 'O', 'U']
    queue = deque([''])
    
    while queue:
        current_word = queue.popleft()
        if len(current_word) < 5:
            for vowel in vowels:
                next_word = current_word + vowel
                queue.append(next_word)
                dictionary.append(next_word)
                
    dictionary.sort()
    return dictionary.index(word) + 1